server/src/test/java/org/kaazing/gateway/server/topic/DefaultClusterTopicTest.java

```diff
@@ -0,0 +1,205 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.kaazing.gateway.server.topic;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.RuleChain;
+import org.kaazing.gateway.server.context.resolve.DefaultClusterContext;
+import org.kaazing.gateway.service.cluster.MemberId;
+import org.kaazing.test.util.ITUtil;
+
+import com.hazelcast.core.ITopic;
+import com.hazelcast.core.MessageListener;
+
+public class DefaultClusterTopicTest {
+
+    private static DefaultClusterContext clusterContext1;
+    private static DefaultClusterContext clusterContext2;
+
+    @Rule
+    public RuleChain chain = ITUtil.createRuleChain(60, TimeUnit.SECONDS);
+
+
+    @BeforeClass
+    public static void startContexts() {
+        MemberId member1 = new MemberId("tcp", "127.0.0.1", 46943);
+        MemberId member2 = new MemberId("tcp", "127.0.0.1", 46942);
+
+        List accepts = Collections.singletonList(member1);
+        List connects = Collections.singletonList(member2);
+        final String clusterName = DefaultClusterTopicTest.class.getName() + "-cluster1";
+        clusterContext1 = new DefaultClusterContext(clusterName,
+            accepts,
+            connects,
+            null);
+
+        clusterContext2 = new DefaultClusterContext(clusterName,
+            connects,
+            accepts,
+            null);
+
+        clusterContext1.start();
+        clusterContext2.start();
+    }
+
+    @AfterClass
+    public static void disposeContexts() {
+        clusterContext1.dispose();
+        clusterContext2.dispose();
+    }
+
+    @Test
+    public void shouldCallMessageListenersOnTwoThreads() throws InterruptedException {
+        ITopic<String> topicMember1 = clusterContext1.getTopic("topic");
+        CountDownLatch listenersCalled = new CountDownLatch(2);
+        topicMember1.addMessageListener(message -> {
+            assertEquals("msg1", message.getMessageObject());
+            listenersCalled.countDown();
+        });
+        Thread t = new Thread(() -> {
+            topicMember1.addMessageListener(message -> {
+                assertEquals("msg1", message.getMessageObject());
+                listenersCalled.countDown();
+            });
+        });
+        t.start();
+        t.join();
+        topicMember1.publish("msg1");
+        listenersCalled.await();
+        assertEquals(1, topicMember1.getLocalTopicStats().getPublishOperationCount());
+        assertEquals(2, topicMember1.getLocalTopicStats().getReceiveOperationCount());
+        topicMember1.destroy();
+    }
+
+    @Test
+    public void shouldSendAndReceiveMessagesBetweenMembers() throws InterruptedException {
```

Comment:
Obviously this method does not apply to the standalone case so would not appear in the abstract superclass.